File: civichub-backend/src/main/java/com/civichub/dashboard/dto/response/StaffDashboardSummaryResponse.java
package com.civichub.dashboard.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class StaffDashboardSummaryResponse {

    private long pendingReports;
    private long receivedReports;
    private long inProgressReports;
    private long resolvedReports;
    private long rejectedReports;
    private long cancelledReports;
    private long totalAssigned;

    public StaffDashboardSummaryResponse(
            Number pendingReports,
            Number receivedReports,
            Number inProgressReports,
            Number resolvedReports,
            Number rejectedReports,
            Number cancelledReports,
            Number totalAssigned) {
        this.pendingReports = toLong(pendingReports);
        this.receivedReports = toLong(receivedReports);
        this.inProgressReports = toLong(inProgressReports);
        this.resolvedReports = toLong(resolvedReports);
        this.rejectedReports = toLong(rejectedReports);
        this.cancelledReports = toLong(cancelledReports);
        this.totalAssigned = toLong(totalAssigned);
    }

    private long toLong(Number value) {
        return value == null ? 0L : value.longValue();
    }
}

File: civichub-mobile/lib/features/staff/data/models/staff_dashboard_summary_response.dart
import '../../domain/models/staff_dashboard_summary.dart';

class StaffDashboardSummaryResponse {
  const StaffDashboardSummaryResponse({
    required this.pendingReports,
    required this.receivedReports,
    required this.inProgressReports,
    required this.resolvedReports,
    required this.rejectedReports,
    required this.cancelledReports,
    required this.totalAssigned,
  });

  final int pendingReports;
  final int receivedReports;
  final int inProgressReports;
  final int resolvedReports;
  final int rejectedReports;
  final int cancelledReports;
  final int totalAssigned;

  factory StaffDashboardSummaryResponse.fromJson(Map<String, dynamic> json) {
    return StaffDashboardSummaryResponse(
      pendingReports: _int(json['pendingReports']),
      receivedReports: _int(json['receivedReports']),
      inProgressReports: _int(json['inProgressReports']),
      resolvedReports: _int(json['resolvedReports']),
      rejectedReports: _int(json['rejectedReports']),
      cancelledReports: _int(json['cancelledReports']),
      totalAssigned: _int(json['totalAssigned']),
    );
  }

  StaffDashboardSummary toDomain() {
    return StaffDashboardSummary(
      pendingReports: pendingReports,
      receivedReports: receivedReports,
      inProgressReports: inProgressReports,
      resolvedReports: resolvedReports,
      rejectedReports: rejectedReports,
      cancelledReports: cancelledReports,
      totalAssigned: totalAssigned,
    );
  }

  static int _int(Object? value) {
    if (value is int) {
      return value;
    }
    if (value is num) {
      return value.toInt();
    }
    return 0;
  }
}
